src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -966,7 +966,11 @@ def export_zone(cmd, resource_group_name, zone_name):
 
             zone_obj[record_set_name][record_type].append(record_obj)
 
-    print(make_zone_file(zone_obj))
+    zone_file = make_zone_file(zone_obj)
+    print(zone_file)
+    if file_name:
+        with open(file_name, 'w') as f:
+            f.write(zone_file)
```

Comment:
I would recommend to capture `IOException` can translate it to a user-friendly error message.